public/apps/configuration/panels/tenant-list/tenant-list.tsx

```diff
@@ -0,0 +1,371 @@
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import {
+  EuiBadge,
+  EuiButton,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiLink,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentHeader,
+  EuiPageContentHeaderSection,
+  EuiPageHeader,
+  EuiPopover,
+  EuiText,
+  EuiTitle,
+  EuiGlobalToastList,
+  EuiOverlayMask,
+  EuiConfirmModal,
+} from '@elastic/eui';
+import React, { ReactNode, useEffect, useState, useCallback } from 'react';
+import { difference } from 'lodash';
+import { getAuthInfo } from '../../../../utils/auth-info-utils';
+import { AppDependencies } from '../../../types';
+import { Action, Tenant } from '../../types';
+import { renderCustomization } from '../../utils/display-utils';
+import {
+  fetchTenants,
+  transformTenantData,
+  fetchCurrentTenant,
+  resolveTenantName,
+  updateTenant,
+  requestDeleteTenant,
+  selectTenant,
+} from '../../utils/tenant-utils';
+import { getNavLinkById } from '../../../../services/chrome_wrapper';
+import { TenantEditModal } from './edit-modal';
+import { useToastState } from '../../utils/toast-utils';
+
+const PAGEID_DICT: { [key: string]: string } = {
+  DashboardId: 'dashboards',
+  VisualizationId: 'visualize',
+};
+
+export function TenantList(props: AppDependencies) {
+  const [tenantData, setTenantData] = useState<Tenant[]>([]);
+  const [errorFlag, setErrorFlag] = useState(false);
+  const [selection, setSelection] = useState<Tenant[]>([]);
+  const [isActionsPopoverOpen, setActionsPopoverOpen] = useState(false);
+  const [currentTenant, setCurrentTenant] = useState('');
+  const [currentUsername, setCurrentUsername] = useState('');
+  // Modal state
+  const [editModal, setEditModal] = useState<ReactNode>(null);
+  const [toasts, addToast, removeToast] = useToastState();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const closeModal = () => setIsModalVisible(false);
+  const showModal = () => setIsModalVisible(true);
+
+  const fetchData = useCallback(async () => {
+    try {
+      const rawTenantData = await fetchTenants(props.coreStart.http);
+      const processedTenantData = transformTenantData(rawTenantData);
+      const activeTenant = await fetchCurrentTenant(props.coreStart.http);
+      setCurrentTenant(resolveTenantName(activeTenant, currentUsername));
+      setTenantData(processedTenantData);
+    } catch (e) {
+      console.log(e);
+      setErrorFlag(true);
+    }
+  }, [currentUsername, props.coreStart.http]);
+
+  useEffect(() => {
+    const fetchCurrentUser = async () => {
```

Comment:
currentUser seems not special from other data inside fetchData. Probably merge it into the fetchData for simplicity.

Reply:
Right. merged.